src/v2/stack_aggregator.py

```diff
@@ -309,13 +312,18 @@ def initiate_unknown_package_ingestion(self):
             return
 
         ecosystem = self._normalized_packages.ecosystem
+        pkg_list = self.get_all_unknown_packages()
         try:
-            for dep in self.get_all_unknown_packages():
-                server_create_analysis(ecosystem, dep.name, dep.version, api_flow=True,
-                                       force=False, force_graph_sync=True)
-        except Exception as e:  # pylint:disable=W0703,C0103
-            logger.error('Ingestion failed for {%s, %s, %s}',
-                         ecosystem, dep.name, dep.version)
+            if pkg_list:
+                # Converting tuple keys as utils expects different key names
+                pkg_list_temp = set()
+                for pkg in pkg_list:
+                    pkg_list_temp.add(_PACKAGE(package=pkg.name, version=pkg.version))
+
+                logger.info('Initializing unknown pakage ingestion for eco {} and Packages {}.'
+                            .format(ecosystem, pkg_list))
+                unknown_package_flow(ecosystem, pkg_list_temp)
```

Comment:
- Use either map or list comprehension. 
- Don't use preformated string for logging
- Follow naming convention
```suggestion
unknown_pkgs = map(lambda pkg: (pkg.name, pkg.version), pkg_list)
unknown_package_flow(ecosystem, unknown_pkgs)
logger.debug('Unknown ingestion executed for %s packages in %s ecosystem', len(pkg_list), ecosystem)
```

Reply:
@jparsai where are we filtering out the packages which might be private in nature?

Reply:
@yzainee as of now I have validation for pseudo version in ingestion API, for private packages I will put check in worker.